flow/connectors/postgres/cdc.go

```diff
@@ -340,9 +342,15 @@ func (p *PostgresCDCSource) consumeStream(
 		} else {
 			ctx, cancel = context.WithDeadline(p.ctx, nextStandbyMessageDeadline)
 		}
-
 		rawMsg, err := conn.ReceiveMessage(ctx)
 		cancel()
+
+		utils.RecordHeartbeatWithRecover(p.ctx, "consumeStream ReceiveMessage")
```

Comment:
This is put here to improve speed at which we react to context cancellation, since the error handling below is looking at `p.commitLock` but our context being cancelled isn't something we can ignore